src/firewall/core/io/firewalld_conf.py

```diff
@@ -10,9 +10,53 @@
 import tempfile
 import shutil
 
+import firewall.errors
+import firewall.functions
 from firewall import config
 from firewall.core.logger import log
 
+
+def _validate_bool(value, default):
+    valid = True
+    try:
+        v = firewall.functions.str_to_bool(value, default=_validate_bool)
+    except ValueError:
+        valid = False
+        v = _validate_bool
+
+    if v is _validate_bool:
+        v = firewall.functions.str_to_bool(default)
```

Comment:
With my other feedback this should be simplified as well. To a single `str_to_bool()` call.

Reply:
Your other feedback strictly takes *away* functionality from `str_to_bool()`. So, if with your feedback the code could be replaced by a single `str_to_bool()` call, then it also could without that change.

The function is that way, because it aims to detect whether the value is valid and indicate that via a return value. `str_to_bool()` doesn't return two values, instead it indicates invalid input with an exception. Having the caller of `_validate_bool()` add a try-except is inconvenient (for the caller). Hence, `_validate_bool()` is more complicated.

Also, `str_to_bool()` does not try to parse the default value, it returns it directly. Here, `_validate_bool()` also parses the default value to give a boolean value. You couldn't do that with a single `str_to_bool()` call, unless you let `str_to_bool()` try to also parse the default value as `True`/`False`, which would take other functionality away from `str_to_bool()`.